### ISSUE TITLE
Tips for Gemma 3 1B pretraining

### ISSUE THREAD

**Issue by Kurokabe** [NVIDIA/NeMo#13438]
Hi,

I'm attempting to pretrain the **Gemma 3 1B** model using the new **NeMo `AutoModel`** on a custom dataset. The model successfully overfits on a small subset of the data, as shown below:

![Overfitting on small dataset](https://github.com/user-attachments/assets/ab1a57d6-68b8-4d49-8430-1ed28ce0de11)

However, when I scale up to the full dataset, the training gets stuck at a relatively high loss (around 9) and doesn’t improve:

![Stuck at high loss on full dataset](https://github.com/user-attachments/assets/7f48d710-b7a1-4495-8522-d4c0bbbd3b3d)

For comparison, when I pretrain **LLaMA 3.2 1B** directly using the standard NeMo/Megatron training scripts (not via `AutoModel`), I do not encounter this issue:

![Successful training with LLaMA 3.2 1B](https://github.com/user-attachments/assets/e65e7f90-4974-46e3-8f37-3d1b0dab7cec)

---

Are there any special considerations or tips for pretraining **Gemma 3 models** with NeMo, especially when using `AutoModel`?

I found that older Gemma models [require prepending a `<bos>` token](https://unsloth.ai/blog/gemma-bugs) (which I did) to avoid having a large starting loss. Is there any other thing to be careful with Gemma 3?

Any insights would be greatly appreciated.

Thanks!

**Comment by akoumpa** [comment]
Hi @Kurokabe,

thanks for raising. I was able to reproduce the issue, I think there might be some issue with the initialization of the model. I'm looking into it and hope to have an update soon.

thank you.
